src/idpLoader.ts

```diff
@@ -0,0 +1,91 @@
+import * as https from "https";
+import { DOMParser } from "xmldom";
+import { log } from "./utils/logger";
+
+interface IDPMetadata {
+  cert: ReadonlyArray<string>;
+  entityID: string;
+  entryPoint: string;
+  logoutUrl: string;
+}
+
+const EntityDescriptorTAG = "md:EntityDescriptor";
+const X509CertificateTAG = "ds:X509Certificate";
+const SingleSignOnServiceTAG = "md:SingleSignOnService";
+const SingleLogoutServiceTAG = "md:SingleLogoutService";
+
+export async function parseIdpMetadata(): Promise<ReadonlyArray<IDPMetadata>> {
+  const IDP_METADATA_ULR =
+    process.env.IDP_METADATA_ULR ||
+    "https://registry.spid.gov.it/metadata/idp/spid-entities-idps.xml"; // TODO: remove default configuration?
+  const ipdMetadataPage = await getRequest(IDP_METADATA_ULR); // TODO: Failure on download remote configuration will throw an exception
+  const domParser = new DOMParser().parseFromString(ipdMetadataPage);
+  const entityDescriptors = domParser.getElementsByTagName(EntityDescriptorTAG);
+  return Array.from(entityDescriptors).reduce(
+    (idp: ReadonlyArray<IDPMetadata>, element: Element) => {
+      if (validateEntityDescriptorFormat(element)) {
+        const elementInfo = {
+          cert: [
+            (element
+              .getElementsByTagName(X509CertificateTAG)
+              .item(0) as Element).textContent as string
```

Comment:
there can be more than one certificate

Reply:
also, what if there are no certificates in the xml?

Reply:
At this time only the first will be taken. If no certificate is present the `IDPEntityDescriptor.decode` will fail and the IDP will be skipped.
I'll add code to handle multiple certificates

Reply:
fixed on commit d9bf01c

Reply:
with commit 9aa766b I'll add `minItems` on `IDPEntityDescriptor.cert` but it seems that the generator doesn't support that.

Reply:
fixed on commit baa27f1, now empty array are not accepted